### DISCUSSION TITLE
Stitches source code from TS to JS?

### DISCUSSION THREAD

**Discussion by baryla** [stitchesjs/stitches#391] [Ideas]
Hi there! 

I've been following Stitches since the very first announcement and the recent changes to the repo raised my eyebrows a little bit, hence this discussion. It's not my place to say how Stitches should be developed (of course) but I'm just curious why the change from TS to JS for the source code? For me personally (as a potential contributor), it makes the code more difficult to read and arguably harder to maintain since the code is not typed.

Also, I think this repo would benefit from a `prettier` config or at least an update to `.editorconfig` to enforce the indentation style for all files to be "spaces" because it breaks the formatting when previewing files on Github (I can create a PR for this if you're open to that?).

Keep up the great work everyone 👊

**Comment by graup** [comment]
+1. I'd like an answer to this. As someone checking out this project and starting to get into the code, that's quite a turnoff.

**Comment by LucasUnplugged** [comment]
Very interesting. I didn't know the history here, but find myself wishing the code base was in TS, so I'm also curious why the devs would have switched.

In general, I find the barrier to entry for TS fairly small (outside of rare, super complex cases), since TS is a superset of JS — much as that makes the barrier to entry low for SCSS, for folks who only know CSS. And only having to make small changes to a code base (e.g., via small PRs) makes that even easier, compared to having to know TS well enough to use it from scratch.

I went through that exact experience only 14 months ago, and already I would never start a code base in vanilla JS again, if I had a choice.

**Comment by rafgraph** [comment]
The biggest benefit I’ve gotten from code written in TS over the years is that it makes unfamiliar code bases *much* easier to work in. So to not have an open source project written in TS is definitely a turnoff as a potential contributor.
